src/coreclr/jit/gentree.cpp

```diff
@@ -27546,12 +27546,13 @@ bool GenTreeHWIntrinsic::OperIsMemoryLoad(GenTree** pAddr) const
     {
 #ifdef TARGET_ARM64
         static_assert_no_msg(
-            AreContiguous(NI_Sve_GatherVector, NI_Sve_GatherVectorByteZeroExtend, NI_Sve_GatherVectorInt16SignExtend,
+            AreContiguous(NI_Sve_GatherPrefetch16Bit, NI_Sve_GatherPrefetch32Bit, NI_Sve_GatherPrefetch64Bit, NI_Sve_GatherPrefetch8Bit,
```

Comment:
I don't think the prefetches need to be listed as memory loads, given they don't actually load memory (they just hint to the processor that we might access this address).

We don't mark cases like `NI_SSE_Prefetch0` on xarch as loads, for example, we just rather track it as `HW_Flag_SpecialSideEffect_Other`

Reply:
i will revert this change for now and have a separate PR to fix it properly.